image_services/opera/make_opera_services.py

```diff
@@ -18,8 +19,17 @@
 gdal.UseExceptions()
 gdal.SetConfigOption('GDAL_DISABLE_READDIR_ON_OPEN', 'EMPTY_DIR')
 
-def get_rasters():
-    url_file = f'{os.getcwd()}/vsis3_urls.txt'
+
+def get_rasters(overview_path, s3_suffix, working_directory):
+    _, _, bucket, _, _ = overview_path.split('/')
+    polarization = s3_suffix[1:3]
+    filename = f'{working_directory}/opera_vsis3_{polarization}.csv'
+    key =  f'opera_uris/opera_vsis3_{polarization}.csv'
+
+    s3 = boto3.client('s3')
+    logging.info(f'Downloading {bucket}/{key} to {filename}')
+    url_file = s3.download_file(bucket, key, filename)
+
     with open(url_file, newline='') as urlfile:
```

Comment:
[download_file()](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/download_file.html) doesn't have a return value, so `url_file` ends up as `None` here. You should be able to pass `filename` to the subsequent `open()` call:

```suggestion
    s3.download_file(bucket, key, filename)

    with open(filename, newline='') as urlfile:
```